src/python/pants/backend/python/subsystems/debugpy.py

```diff
@@ -29,6 +32,37 @@ class DebugPy(PythonToolBase):
     default_lockfile_path = "src/python/pants/backend/python/subsystems/debugpy.lock"
     default_lockfile_url = git_url(default_lockfile_path)
 
+    args = ArgsListOption(example="--log-to-stderr")
+
+    @staticmethod
+    def _get_main_spec_args(main: MainSpecification) -> tuple[str, ...]:
+        if isinstance(main, EntryPoint):
+            if main.function:
+                return ("-c", f"import {main.module};{main.module}.{main.function}();")
+            return ("-m", main.module)
+
+        assert isinstance(main, ConsoleScript)
```

Comment:
It seems like this method should take a `ConsoleScript` so that the caller is forced to apply (and reason about) this assertion instead.

Reply:
Well either entrypoint or console script, I'll change the type to a union

Reply:
Actually, I think I'm going to leave it as-is. These are the only two subclasses of `MainSpecification`, and if I change it to use the union, the callsite needs to be casted or asserted, and neither is pretty.

Reply:
The assert is still happening: it's just hidden. IMO, that's a trap for the caller. But nbd.

Reply:
Yeah OK, thats fair. Past Josh was lazy, we should shun him.